tests/unittests/tests-checksum/tests-checksum-fletcher16.c

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2016 Ludwig Knüpfer <ludwig.knuepfer@fu-berlin.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+#include <stdint.h>
+
+#include "embUnit/embUnit.h"
+
+#include "checksum/fletcher16.h"
+
+#include "tests-checksum.h"
+
+static int calc_and_compare_checksum(const unsigned char *buf, size_t len,
+                                     uint16_t expected)
+{
+    uint16_t result = fletcher16(buf, len);
+
+    return result == expected;
+}
+
+static void test_checksum_fletcher16(void)
+{
+    {
+        unsigned char buf[] = "";
+        uint16_t expect = 0xFFFF;
+
+        TEST_ASSERT(calc_and_compare_checksum(buf, sizeof(buf) - 1, expect));
+    }
```

Comment:
Syntax is C I hope.

The function has several asserts because it tests several values.